### DISCUSSION TITLE
SSHOperator's logs severity are not coherent to the actual exception severity

### DISCUSSION THREAD

**Discussion by medina325** [apache/airflow#25329] [Q&A]
Hello Airflow Community!

### Apache Airflow version
2.3.3 (latest released)

### Deployment
docker-compose

### What happened

Our current client requires specific DAGs to be run in another machine, so we've been using the SSHOperator to get the job done.

Everything was good until the client reported the DAG was not generating the expected results, which we found weird since every task sends notifications on failures (which was working fine in other situations, e.g.: where the connetion failed).

We checked and found out the Scrapy application being run was actually throwing exceptions (ERROR level) but the SSHOperator was just logging them as WARNINGs.

![image](https://user-images.githubusercontent.com/56964034/181143492-732efa20-80b9-4218-94c3-2a0c4cd1ddf0.png)

This is a very serious problem since we have no way to break the DAG when our application throws exceptions, since the DAG is simply gonna be unware of them.

### Anything else
I know about [this discussion](https://github.com/apache/airflow/discussions/22515), where the exit code of the bash command was not equal to 0. I googled a bit and found this [Scrapy issue](https://github.com/scrapy/scrapy/issues/1231) which talks about the exit code of Scrapy not being 0 when a Spider (web crawler) breaks, therefore the SSHOperator is not considered the task as failed.

### What you think should happen instead
The SSHOperator's logs level should match the application's logs.

Is there anything I could do to get Scrapy's log severity?

**Comment by potiuk** [comment]
Our operator should not rely on logging levels. You need to make sure your script returns non-0 value when it experiences an error,. You can do it for example by writing a custom shell script (on the remote system) around scrappy to parse the logs and exit with non-0 status when scrappy prints ERROR. This is very specific to your case and writing customized script that converts a "specific" scrappy way of reporting problems to the way that Airlfow SSH Operator expects (exit coce) is the only way to proceed.

**Reply by medina325** [reply]
Yeah, I understand what you are saying makes sense. I'm gonna see how I can wrap the ``scrapy crawl`` command in a custom script, any ideas/suggestions? hehe

**Reply by potiuk** [reply]
Hehe. No.
